src/daffodilDebugger.ts

```diff
@@ -6,76 +6,80 @@ import * as child_process from 'child_process';
-                const stream = response.message.pipe(file);
-                stream.on("close", () => {
-                    try { res(filePath); } catch (err) { rej(err); }
+                await new Promise((res, rej) => {
+                    file.on("error", (err) => function () { throw err });
+                    const stream = response.message.pipe(file);
+                    stream.on("close", () => {
+                        try { res(filePath); } catch (err) { rej(err); }
+                    });
                 });
-            });
 
-            // Unzip file and remove zip file
-            await new Promise ((res, rej) => { 
-                let stream = fs.createReadStream(filePath).pipe(unzip.Extract({ path: `${rootPath}` }));
-                stream.on("close", () => {
-                    try { res(filePath); } catch (err) { rej(err); }
+                // Unzip file and remove zip file
+                await new Promise ((res, rej) => { 
+                    let stream = fs.createReadStream(filePath).pipe(unzip.Extract({ path: `${rootPath}` }));
+                    stream.on("close", () => {
+                        try { res(filePath); } catch (err) { rej(err); }
+                    });
                 });
-            });
-            fs.unlinkSync(filePath);
-        }
+                fs.unlinkSync(filePath);
+            }
 
-        // Stop debugger if running
-        if (os.platform() === 'win32') {
-            // Windows stop debugger if already running
-            child_process.exec("tskill java");
-        }
-        else {
-            // Linux/Mac stop debugger if already running and make sure script is executable
-            child_process.exec("kill -9 $(ps -ef | grep 'daffodil' | grep 'jar' | awk '{ print $2 }') || return 0") // ensure debugger server not running and
-            child_process.exec(`chmod +x ${rootPath}/daffodil-debugger-${dapodilDebuggerVersion.substring(1)}/bin/da-podil`)     // make sure da-podil is executable
-        }
+            // Stop debugger if running
+            if (os.platform() === 'win32') {
+                // Windows stop debugger if already running
+                child_process.exec("tskill java");
+            }
+            else {
+                // Linux/Mac stop debugger if already running and make sure script is executable
+                child_process.exec("kill -9 $(ps -ef | grep 'daffodil' | grep 'jar' | awk '{ print $2 }') || return 0") // ensure debugger server not running and
+                child_process.exec(`chmod +x ${rootPath}/daffodil-debugger-${dapodilDebuggerVersion.substring(1)}/bin/da-podil`)     // make sure da-podil is executable
+            }
 
-        // Assign script name based on os platform
-        let scriptName = os.platform() === 'win32' ? "da-podil.bat": "da-podil";
+            // Assign script name based on os platform
+            let scriptName = os.platform() === 'win32' ? "da-podil.bat": "da-podil";
```

Comment:
On Linux the command here will (most likely) fail due to the script not being on the path.  Making it `./da-podil` is a quick fix, another way would be to pass an environment map to `createTerminal` that sets the `PATH`.